pkg/source/source_test.go

```diff
@@ -161,6 +162,16 @@ func TestFetchSourceFromMelange(t *testing.T) {
 					t.Errorf("Expected step %s, got %s", tc.expectedSteps[i], step)
 				}
 			}
+
+			// Validate the files in the destination directory
+			if tc.expectedFiles != nil {
```

Comment:
you might consider:
```go
if tc.expectedFiles == nil {
  continue
}
```
to get rid of a level of indention (we're pretty nested here already)